### ISSUE TITLE
mapStateToProps() in Connect(Header) and Connect(Post) must return a plain object. Instead received undefined.

### ISSUE THREAD

**Issue by anandp926** [anandp926/reabable#1]
Hi, I tried to remove this mapStateToProps() in Connect(Header) and Connect(Post) must return a plain object. Instead received undefined. error but not find any suitable solution.
could anyone help me, please?

@Andarist

reducers/Post.js

/**
 * Created by rozer on 4/5/2018.
 */
import { GET_POSTS } from '../actions/posts'

function posts(state = {}, action) {
    switch (action.type){
        case GET_POSTS:
            return action.posts;
        default:
            return state;
    }
}

export default posts

actions/Post.js

/**
 * Created by rozer on 4/5/2018.
 */
import * as api from '../utils/api'

export const GET_POSTS = 'GET_POSTS';

export const getAllPosts = posts => ({
    type: GET_POSTS,
    posts
});

export const fetchAllPosts = () => (dispatch) =>(
    api.getAllPosts().then(posts => dispatch(getAllPosts(posts)))
);

component/Post.js

/**
 * Created by rozer on 3/31/2018.
 */
import React, { Component } from 'react'
import FaThumbsOUp from 'react-icons/lib/fa/thumbs-o-up'
import FaCommentO from 'react-icons/lib/fa/comment-o'
import { connect } from 'react-redux'
import { fetchAllPosts } from '../actions/posts'

class Post extends Component {

    componentDidMount(){
        this.props.fetchAllPosts();
    }

    render() {
        console.log(this.props);
        return (
            <div className="card">
                <div className="container">
                    <div className="Time">
                        6:35 PM, Saturday, 31 March 2018
                    </div>
                    <h4>
                        <b><a href="#post">'We are back': Emotional Dhoni on Chennai Super Kings' IPL return</a></b>
                        <small className="Author">By: Anand Singh</small>
                    </h4>
                    <p>
                        NEW DELHI: 'Captain Cool' Mahendra Singh Dhoni got emotional while speaking about the return of Chennai Super Kings (CSK) to the Indian Premier League (IPL) after two years.
                        "We are back! We are back as a whole team," an emotional Dhoni was seen speaking in a brief video shared on Twitter.
                    </p>
                    <div className="Like-comment">
                        <span className="Like-comment1">100 Likes</span>
                        <span className="Like-comment2">564 Comments</span>
                    </div>
                    <div className="Vote-comment">
                        <button className="Like">
                            <FaThumbsOUp size={30}/>Like
                        </button>
                        <button className="Comment">
                            <FaCommentO size={30}/>Comment
                        </button>
                    </div>
                    <div className="Show-comment">
                        I'm working with you for money. if i have money no one can defeat me.
                        <div className="Show-comment-button">
                            <button>replay</button>
                            <button>edit</button>
                            <button>delete</button>
                        </div>
                    </div>
                </div>
            </div>
        );
    }
}

const mapStateToProps = (state) => {
    return {
    posts: state.posts
}};

export default connect(mapStateToProps, fetchAllPosts)(Post);

**Comment by Andarist** [comment]
Hi there 👋 

Was I supposed to be mentioned here? 

Anyway - it seems that you should do this:
```js
export const mapDispatchToProps = (dispatch) =>({
  fetchAllPosts: () => api.getAllPosts().then(posts => dispatch(getAllPosts(posts)))
});
```

I would advise moving actual fetching to a middleware though, or at least do it in the component itself. Side-effects in `mapDispatchToProps` sounds like a code smell to me

**Comment by anandp926** [comment]
sorry, @Andarist I saw your answers in another post and then I watched your git profile. you have a lot of experience.I know this is not the right thing but I was trying to remove the issue from my code since three days. hence to get the fast and accurate answer, I mentioned you. sorry about that.

As you said i changed my code now i am getting this error:

The previous state received by the reducer has unexpected type of "Function". Expected argument to be an object with the following keys: "categories", "posts"
Header.js:18 
{categories: {…}, fetchCategories: ƒ}
categories
:
{}
fetchCategories
:
ƒ fetchCategories()
__proto__
:
Object
